sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullExpressions.scala

```diff
@@ -357,7 +358,7 @@ case class AtLeastNNonNulls(n: Int, children: Seq[Expression]) extends Predicate
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val nonnull = ctx.freshName("nonnull")
```

Comment:
Does it really matter not to have `nonnull` as a global variable? The code is simpler if it is a global one.

Reply:
I think it does, because having it as a global variable may put more pressure on the constant pool (see SPARK-18016). Thus, whenever feasible, I do think that we should keep it local.